pkg/cloudprovider/cloudprovider.go

```diff
@@ -166,6 +167,11 @@ func getCABundle(ctx context.Context, restConfig *rest.Config) (*string, error)
 }
 
 func kubeDNSIP(ctx context.Context, kubernetesInterface kubernetes.Interface) (net.IP, error) {
+	envIp := os.Getenv("KARPENTER_KUBERNETES_DNS_IP")
```

Comment:
Can we expose this via the recently introduced Settings configmap? This is where all other env vars are defined. It has the benefit of hot reloading without restarting karpenter.